Source/ARTRest.m

```diff
@@ -735,18 +732,17 @@ - (ARTLocalDevice *)device_nosync {
     //
     // As a side effect, the first instance "wins" at setting the device's
     // client ID.
-
-    static dispatch_once_t once;
-    device_once_token = &once;
-    static id device;
-    dispatch_once(&once, ^{
-        device = [ARTLocalDevice load:self.auth.clientId_nosync storage:self.storage];
-    });
-    return device;
+    if (ARTLocalDevice.shared_nosync.id == nil) {
```

Comment:
Because `ARTLocalDevice` should have its own queue for accessing storage, I left it there. `[ARTRest device_nosync]` is just a wrapper accessor. If and when this PR new API accepted, I'll remove usage of this property everywhere and update tests.